application_sandbox/conservative_rasterization/main.cpp

```diff
@@ -0,0 +1,359 @@
+        0, VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL};
+
+    render_pass_ = containers::make_unique<vulkan::VkRenderPass>(
+        data_->allocator(),
+        app()->CreateRenderPass(
+            {{
+                0,                                         // flags
+                render_format(),                           // format
+                num_samples(),                             // samples
+                VK_ATTACHMENT_LOAD_OP_CLEAR,               // loadOp
+                VK_ATTACHMENT_STORE_OP_STORE,              // storeOp
+                VK_ATTACHMENT_LOAD_OP_DONT_CARE,           // stenilLoadOp
+                VK_ATTACHMENT_STORE_OP_DONT_CARE,          // stenilStoreOp
+                VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL,  // initialLayout
+                VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL   // finalLayout
+            }},  // AttachmentDescriptions
+            {{
+                0,                                // flags
+                VK_PIPELINE_BIND_POINT_GRAPHICS,  // pipelineBindPoint
+                0,                                // inputAttachmentCount
+                nullptr,                          // pInputAttachments
+                1,                                // colorAttachmentCount
+                &color_attachment,                // colorAttachment
+                nullptr,                          // pResolveAttachments
+                nullptr,                          // pDepthStencilAttachment
+                0,                                // preserveAttachmentCount
+                nullptr                           // pPreserveAttachments
+            }},                                   // SubpassDescriptions
+            {}                                    // SubpassDependencies
+            ));
+
+    conservative_rasterization_pipeline_ =
+        containers::make_unique<vulkan::VulkanGraphicsPipeline>(
+        data_->allocator(),
+        app()->CreateGraphicsPipeline(pipeline_layout_.get(),
+                                      render_pass_.get(), 0));
+    conservative_rasterization_pipeline_->AddShader(VK_SHADER_STAGE_VERTEX_BIT,
+                                                    "main",
+                              cube_vertex_shader);
+    conservative_rasterization_pipeline_->AddShader(
+        VK_SHADER_STAGE_FRAGMENT_BIT, "main",
+                              cube_fragment_shader);
+    conservative_rasterization_pipeline_->SetTopology(
+        VK_PRIMITIVE_TOPOLOGY_TRIANGLE_LIST);
+    conservative_rasterization_pipeline_->SetInputStreams(&cube_);
+    conservative_rasterization_pipeline_->SetViewport(viewport());
+    conservative_rasterization_pipeline_->SetScissor(scissor());
+    conservative_rasterization_pipeline_->SetSamples(num_samples());
+    conservative_rasterization_pipeline_->AddAttachment();
+
+	// Setup conservative rasterization
+    PFN_vkGetPhysicalDeviceProperties2KHR vkGetPhysicalDeviceProperties2KHR =
```

Comment:
You can add vkGetPhysicalDeviceProperties2KHR to  `vulkan_wrapper/function_table.h`, which would simplify this.
(It wont get resolved unless you use it, so it should be safe)

Reply:
Thanks for letting me know. I will move it to `vulkan_wrapper/function_table.h`